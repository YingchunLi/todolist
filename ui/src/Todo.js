import React, {Component} from 'react';

import * as Actions from './actions';
import {Link} from 'react-router-dom';
import {connect} from "react-redux";

import AppBar from 'material-ui/AppBar';
import Paper from 'material-ui/Paper';
import {Toolbar, ToolbarGroup, ToolbarTitle} from 'material-ui/Toolbar';
import {
  Table,
  TableBody,
  TableHeader,
  TableHeaderColumn,
  TableRow,
  TableRowColumn,
} from 'material-ui/Table';

const TodoDetailTable = ({title}) =>
  <Paper>
    <Toolbar>
      <ToolbarTitle text={title} />
    </Toolbar>
  </Paper>;

class Todo extends Component {
  render() {
    const {match} = this.props;
    return (
      <div>
        <AppBar title="Todo detial" showMenuIconButton={false}/>

        <TodoDetailTable title={`todo ${match.params.id}`} />
      </div>
    );
  }
}

Todo.propTypes = {};
Todo.defaultProps = {};

export default Todo;
