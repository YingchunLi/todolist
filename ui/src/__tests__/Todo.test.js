import React from "react";
import {MemoryRouter, Route} from "react-router-dom";

import {shallow, mount, render} from "enzyme";

import Todo from "../Todo";

describe("Todo", () => {
    it("maps to /todos", () => {
      const wrapper = mount(
        <MemoryRouter initialEntries={[`/todos/1`]}>
          <switch>
            <Route path="/todos/:id" component={Todo}/>
          </switch>
        </MemoryRouter>
      );

      expect(wrapper.find(Todo).exists()).toBe(true);
      expect(wrapper.find(Todo).props().match.params.id).toBe('1');
    });

    it("has a title", () => {
      const wrapper = shallow(
        <Todo match={{params: {id: 1}}}/>
      );

      expect(wrapper.find('h2').length).toBe(1);
      expect(wrapper.find('h2').text()).toBe(`This is the todo page 1`);
    });

  }
);

