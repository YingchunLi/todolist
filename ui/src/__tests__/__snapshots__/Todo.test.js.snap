// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo matches snapshot 1`] = `
<div>
  <div
    className={undefined}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#00bcd4",
        "borderRadius": "0px",
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "flex",
        "fontFamily": "Roboto, sans-serif",
        "muiPrepared": true,
        "paddingLeft": 24,
        "paddingRight": 24,
        "position": "relative",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": "100%",
        "zIndex": 1100,
      }
    }
  >
    <h1
      onClick={[Function]}
      style={
        Object {
          "boxFlex": 1,
          "color": "#ffffff",
          "flex": "1",
          "fontSize": 24,
          "fontWeight": 400,
          "height": 64,
          "letterSpacing": 0,
          "lineHeight": "64px",
          "margin": 0,
          "muiPrepared": true,
          "overflow": "hidden",
          "paddingTop": 0,
          "textOverflow": "ellipsis",
          "whiteSpace": "nowrap",
        }
      }
    >
      Todo detial
    </h1>
    <div
      style={
        Object {
          "marginLeft": "auto",
          "marginRight": -16,
          "marginTop": 8,
          "muiPrepared": true,
        }
      }
    >
      <div>
        <button
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "border": 10,
              "borderRadius": 2,
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 36,
              "lineHeight": "36px",
              "margin": 0,
              "minWidth": 88,
              "muiPrepared": true,
              "outline": "none",
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "verticalAlign": null,
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <svg
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "display": "inline-block",
                  "fill": "rgba(0, 0, 0, 0.87)",
                  "height": 24,
                  "marginLeft": 12,
                  "marginRight": 0,
                  "muiPrepared": true,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "verticalAlign": "middle",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
            <span
              style={
                Object {
                  "fontSize": 14,
                  "fontWeight": 500,
                  "letterSpacing": 0,
                  "muiPrepared": true,
                  "paddingLeft": 8,
                  "paddingRight": 16,
                  "position": "relative",
                  "textTransform": "uppercase",
                  "verticalAlign": "middle",
                }
              }
            >
              Edit
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#ffffff",
        "borderRadius": 2,
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "fontFamily": "Roboto, sans-serif",
        "muiPrepared": true,
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      }
    }
  >
    <div
      className={undefined}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "rgb(232, 232, 232)",
          "boxSizing": "border-box",
          "display": "flex",
          "height": 56,
          "justifyContent": "space-between",
          "muiPrepared": true,
          "padding": "0px 24px",
        }
      }
    >
      <span
        style={
          Object {
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 20,
            "lineHeight": "56px",
            "muiPrepared": true,
            "overflow": "hidden",
            "paddingRight": 16,
            "position": "relative",
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
          }
        }
      >
        todo 1
      </span>
    </div>
    <div
      style={
        Object {
          "height": "auto",
          "muiPrepared": true,
          "overflow": "auto",
        }
      }
    >
      <div
        style={
          Object {
            "muiPrepared": true,
          }
        }
      >
        <table
          className={undefined}
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderCollapse": "collapse",
              "borderSpacing": 0,
              "fontFamily": "Roboto, sans-serif",
              "tableLayout": "fixed",
              "width": "100%",
            }
          }
        />
      </div>
      <div
        style={
          Object {
            "height": "inherit",
            "muiPrepared": true,
            "overflowX": "hidden",
            "overflowY": "auto",
          }
        }
      >
        <table
          className={undefined}
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderCollapse": "collapse",
              "borderSpacing": 0,
              "fontFamily": "Roboto, sans-serif",
              "tableLayout": "fixed",
              "width": "100%",
            }
          }
        >
          <tbody
            style={
              Object {
                "muiPrepared": true,
              }
            }
          >
            <tr
              className={undefined}
              style={
                Object {
                  "borderBottom": "1px solid #e0e0e0",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "height": 48,
                  "muiPrepared": true,
                }
              }
            >
              <td
                className={undefined}
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "backgroundColor": "inherit",
                    "fontSize": 13,
                    "height": 48,
                    "muiPrepared": true,
                    "overflow": "hidden",
                    "paddingLeft": 24,
                    "paddingRight": 24,
                    "textAlign": "left",
                    "textOverflow": "ellipsis",
                    "whiteSpace": "nowrap",
                    "width": "30%",
                  }
                }
              >
                Name
              </td>
              <td
                className={undefined}
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "backgroundColor": "inherit",
                    "fontSize": 13,
                    "height": 48,
                    "muiPrepared": true,
                    "overflow": "hidden",
                    "paddingLeft": 24,
                    "paddingRight": 24,
                    "textAlign": "left",
                    "textOverflow": "ellipsis",
                    "whiteSpace": "nowrap",
                  }
                }
              />
            </tr>
            <tr
              className={undefined}
              style={
                Object {
                  "borderBottom": "1px solid #e0e0e0",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "height": 48,
                  "muiPrepared": true,
                }
              }
            >
              <td
                className={undefined}
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "backgroundColor": "inherit",
                    "fontSize": 13,
                    "height": 48,
                    "muiPrepared": true,
                    "overflow": "hidden",
                    "paddingLeft": 24,
                    "paddingRight": 24,
                    "textAlign": "left",
                    "textOverflow": "ellipsis",
                    "whiteSpace": "nowrap",
                  }
                }
              >
                Description
              </td>
              <td
                className={undefined}
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "backgroundColor": "inherit",
                    "fontSize": 13,
                    "height": 48,
                    "muiPrepared": true,
                    "overflow": "hidden",
                    "paddingLeft": 24,
                    "paddingRight": 24,
                    "textAlign": "left",
                    "textOverflow": "ellipsis",
                    "whiteSpace": "nowrap",
                  }
                }
              />
            </tr>
            <tr
              className={undefined}
              style={
                Object {
                  "borderBottom": "1px solid #e0e0e0",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "height": 48,
                  "muiPrepared": true,
                }
              }
            >
              <td
                className={undefined}
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "backgroundColor": "inherit",
                    "fontSize": 13,
                    "height": 48,
                    "muiPrepared": true,
                    "overflow": "hidden",
                    "paddingLeft": 24,
                    "paddingRight": 24,
                    "textAlign": "left",
                    "textOverflow": "ellipsis",
                    "whiteSpace": "nowrap",
                  }
                }
              >
                Due Date
              </td>
              <td
                className={undefined}
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "backgroundColor": "inherit",
                    "fontSize": 13,
                    "height": 48,
                    "muiPrepared": true,
                    "overflow": "hidden",
                    "paddingLeft": 24,
                    "paddingRight": 24,
                    "textAlign": "left",
                    "textOverflow": "ellipsis",
                    "whiteSpace": "nowrap",
                  }
                }
              />
            </tr>
            <tr
              className={undefined}
              style={
                Object {
                  "borderBottom": false,
                  "color": "rgba(0, 0, 0, 0.87)",
                  "height": 48,
                  "muiPrepared": true,
                }
              }
            >
              <td
                className={undefined}
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "backgroundColor": "inherit",
                    "fontSize": 13,
                    "height": 48,
                    "muiPrepared": true,
                    "overflow": "hidden",
                    "paddingLeft": 24,
                    "paddingRight": 24,
                    "textAlign": "left",
                    "textOverflow": "ellipsis",
                    "whiteSpace": "nowrap",
                  }
                }
              >
                Status
              </td>
              <td
                className={undefined}
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "backgroundColor": "inherit",
                    "fontSize": 13,
                    "height": 48,
                    "muiPrepared": true,
                    "overflow": "hidden",
                    "paddingLeft": 24,
                    "paddingRight": 24,
                    "textAlign": "left",
                    "textOverflow": "ellipsis",
                    "whiteSpace": "nowrap",
                  }
                }
              />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div
    className={undefined}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#ffffff",
        "borderRadius": 2,
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "marginTop": 10,
        "minWidth": 88,
        "muiPrepared": true,
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      }
    }
  >
    <button
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "#00bcd4",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": "100%",
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <div
          style={
            Object {
              "backgroundColor": false,
              "borderRadius": 2,
              "height": 36,
              "muiPrepared": true,
              "top": 0,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <span
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "margin": 0,
                "muiPrepared": true,
                "opacity": 1,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "userSelect": "none",
              }
            }
          >
            Go back
          </span>
        </div>
      </div>
    </button>
  </div>
</div>
`;
